src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationException("Can't find user with email " + email);
+        }
+        User user = userFromDbOptional.get();
+        if (HashUtil.hashPassword(password, user.getSalt()).equals(user.getPassword())) {
+            return user;
+        }
+        throw new AuthenticationException("email or password is invalid! For user with email "
+                + email);
```

Comment:
You may combine two checks: whether the user has been found by login and does passwords match. (checklist)